include/seqan/align_split/align_split_interface.h

```diff
@@ -303,78 +312,104 @@ void _reverseTrace(String<TraceSegment_<TPosition, TSize>, TSpec> & trace)
 }
 
 // ----------------------------------------------------------------------------
+// Function _computeSplitTrace()
+// ----------------------------------------------------------------------------
+
+template <typename TTarget,
+          typename TSeqH,
+          typename TSeqV,
+          typename TDPContext,
+          typename TMatPos,
+          typename TDPType, typename TBandSwitch, typename TFreeEndGaps, typename TTraceConfig>
+void _computeSplitTrace(TTarget & target,
+                        TSeqH const & seqH,
+                        TSeqV const & seqV,
+                        TDPContext const & dpContext,
+                        TMatPos const matPos,
+                        AlignConfig2<TDPType, DPBandConfig<TBandSwitch>, TFreeEndGaps, TTraceConfig> const & config)
+
+{
+    typedef typename SetupAlignmentProfile_<TDPType, TFreeEndGaps, LinearGaps, TTraceConfig>::Type TDPProfile;
+
+    typedef typename GetDPTraceMatrix<TDPContext const>::Type TDPTraceMatrixHost;
+    typedef typename Value<TDPTraceMatrixHost>::Type TTraceValue;
+
+    typedef DPMatrix_<TTraceValue, FullDPMatrix> TDPTraceMatrix;
+    typedef DPMatrixNavigator_<TDPTraceMatrix, DPTraceMatrix<TTraceConfig>, NavigateColumnWise> TDPTraceMatrixNavigator;
+
+    TDPTraceMatrix matrix;
+        setLength(matrix, +DPMatrixDimension_::HORIZONTAL, length(seqH) + 1 - std::max(0, lowerDiagonal(config._band)));
+    if (IsSameType<TBandSwitch, BandOff>::VALUE)
+    {
+        setLength(matrix, +DPMatrixDimension_::VERTICAL, length(seqV) + 1);
+    }
+    else
+    {
+        int bandSize = _min(static_cast<int>(length(seqH)), upperDiagonal(config._band)) -
+                       _max(lowerDiagonal(config._band), -static_cast<int>(length(seqV))) + 1;
+        setLength(matrix, +DPMatrixDimension_::VERTICAL, _min(static_cast<int>(length(seqV)) + 1, bandSize));
+    }
+
+    setHost(matrix, getDpTraceMatrix(dpContext));
+    resize(matrix);
+    SEQAN_ASSERT_EQ(length(getDpTraceMatrix(dpContext)), length(matrix));
+    TDPTraceMatrixNavigator navi;
+    _init(navi, matrix, config._band);
+    _computeTraceback(target, navi, matPos, seqH, seqV, config._band, TDPProfile());
+}
+
+// ----------------------------------------------------------------------------
 // Function _splitAlignmentImpl()
 // ----------------------------------------------------------------------------
 
 // We call the long sequence contig and the shorter one read but could be changed roles.
-template <typename TContigSeqL, typename TReadSeqL, typename TContigSeqR, typename TReadSeqR,
+template <typename TContigSeqL, typename TReadSeqL,
```

Comment:
following the other recent style for typename, please put these on individual lines, if the corresponding args are on individual lines, too!